element/element.go

```diff
@@ -0,0 +1,36 @@
+package element
+
+import "strings"
+
+type ElementMatcher struct {
```

Comment:
It might be useful to have constructor for this that can do any pre-processing work you might want to do.